source/agora/cli/multi/main.d

```diff
@@ -0,0 +1,312 @@
+/*******************************************************************************
+
+    Entry point for the Agora multi
+
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.cli.multi.main;
+
+import agora.api.FullNode;
+
+import agora.common.Config;
+import agora.node.Node;
+import agora.utils.Log;
+
+import vibe.core.core;
+import vibe.web.rest;
+
+import std.exception;
+import std.file;
+import std.format;
+import std.path;
+import std.range;
+import std.stdio;
+import std.string;
+
+
+/// Default port for nodes when forwarding ports
+private immutable ushort NODE_BASE_PORT = 4000;
+
+/// Admin port for nodes when forwarding ports
+private immutable ushort ADMIN_BASE_PORT = 5000;
+
+/*******************************************************************************
+
+    Application entry point
+
+*******************************************************************************/
+
+private int main (string[] args)
+{
+    // Validate parameters
+    if ((args.length < 3) || ((args[1] != "start") && (args[1] != "test")))
+    {
+        printDefaultHelp();
+        return 1;
+    }
+
+    string[] files;
+    // If the root path is not set,
+    if (extension(args[2]).toLower == ".yaml")
+        files = args[2..$];
+    else
+    // If the root path is not set,
+        foreach (idx; 3..args.length)
+            files ~= buildNormalizedPath(args[2], args[idx]);
+
+    // Reads all configurations from the file.
+    Config[] configs;
+    foreach (idx, file; files)
+    {
+        try
+        {
+            // Read
+            auto config = readFromFile(file);
+
+            // Convert data to suit the environment.
+            config.node.address = "127.0.0.1";
+            config.node.port = cast(ushort)(NODE_BASE_PORT + idx);
+            config.node.data_dir = buildNormalizedPath(dirName(file), config.node.data_dir);
+            config.admin.port = cast(ushort)(ADMIN_BASE_PORT + idx);
+            config.logging.log_level = LogLevel.Error;
```

Comment:
That might be the problem. Here the port is blindly rewritten without taking consideration what is set in the config file. Why is that ?

Reply:
There's a log of three nodes. It's not easy to recognize, I think we need something new.

Reply:
The comment is more about `port` and `address` than `log_level`.
But in general, we should aim to override as little as possible, because it will surprise the user.